.github/workflows/charm-update-libs.yaml

```
@@ -27,26 +27,87 @@ jobs:
+            # If there is a new major version of the library, open a new issue
+            if [[ $(printf "%s\n%s" "$lib_major" "$latest_major" | sort -V | tail -n1 ) != "$lib_major" ]]; then
+              issue_body=$(printf "%s\n%s" "$issue_body" "- update <b>$lib</b> to <b>$latest_major</b>")
+            fi
+          done
+          # Check if there are already open issues for major library upgrades
+          open_issues_count="$(gh issue list --search 'chore: update libraries to new major versions' --state open --json id --jq 'length')"
+          if [[ "$open_issues_count" == "0" && "$issue_body" != "" ]]; then
+            echo "Creating a GitHub issue for the major library version update"
+            issue_body="$(printf "%s\n\n%s\n%s" "This issue was created automatically because a new major version was detected for a charm library." "You should update the following libraries:" "${issue_body}")"
+            gh issue create \
+              --title "chore: update libraries to new major versions" \
+              --body "$issue_body"
+          fi
+          cd "$GITHUB_WORKSPACE"
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMHUB_TOKEN }}"
+          GH_TOKEN: "${{ secrets.OBSERVABILITY_NOCTUA_TOKEN }}"
+
+      - name: Fetch charm libraries
+        run: |
+          cd "$GITHUB_WORKSPACE/${{ inputs.charm-path }}"
           charmcraft fetch-lib
         env:
           CHARMCRAFT_AUTH: "${{ secrets.CHARMHUB_TOKEN }}"
 
+      - name: Import and configure the GPG key for Noctua
+        uses: crazy-max/ghaction-import-gpg@v6
+        with:
+          gpg_private_key: ${{ secrets.NOCTUA_GPG_PRIVATE }}
+          passphrase: ${{ secrets.NOCTUA_GPG_PASSPHRASE }}
+          git_config_global: true
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+
       - name: Create a PR for local changes
         uses: peter-evans/create-pull-request@v6
         id: cpr
         with:
           token: ${{ secrets.OBSERVABILITY_NOCTUA_TOKEN }}
           commit-message: "chore: update charm libraries"
-          committer: "Github Actions <github-actions@github.com>"
-          author: "Github Actions <github-actions@github.com>"
-          title: "Update charm libraries"
+          committer: "Noctua <webops+observability-noctua-bot@canonical.com>"
+          author: "Noctua <webops+observability-noctua-bot@canonical.com>"
+          title: "chore: update charm libraries"
           body: |
             Automated action to fetch latest version of charm libraries. The branch of this PR 
             will be wiped during the next check. Unless you really know what you're doing, you 
             most likely don't want to push any commits to this branch.
-          branch: "chore/auto-libs"
+
+            The PR will be auto-merged if the CI is green, unless it includes a major library
```

```suggestion
            If all version bumps are minor, and the CI is green, the PR will be auto-merged. If there is a major version bump in any of the libraries, this PR will just sit here until you fix manually all that's breaking.
```

Bit confused:
 I thought we concluded that the chore/auto-libs branch and PR should only include minor bumps. Any major bump will result in an issue, but NOT a pr branch since you're not supposed to be touching that branch (as it will be overwritten at the next check), and the code will likely be broken.

IIRC the flow should be:
```
IF any minor bumps:
  open PR with all minor bumps
  IF CI is green:
    merge
IF any major bumps:
  open issue with all major bumps (or rather actually one per library!!)

---

The flow you describe in the code block is correct, except `IF CI is green: merge` happens before opening the new PR; I just forgot to update this description